dashboard/app/lib/javascripts/dashboard/main.js

```diff
@@ -65,6 +69,87 @@ window.Dashboard = {
 		Marbles.history.navigate('login'+ redirectPath);
 	},
 
+	__catchInsecurePingResponse: function(httpsArgs) {
+		var httpsXhr = httpsArgs[1], self = this,
+			handleSuccess, handleError;
+
+		handleSuccess = function (httpArgs) {
+			var httpXhr = httpArgs[1];
+			// https did not work but http did...something is wrong with the cert
+			self.Dispatcher.handleAppEvent({
+				name: "HTTPS_CERT_MISSING",
+				status: httpXhr.status
+			});
+		};
+		handleError = function (httpArgs) {
+			var httpXhr = httpArgs[1];
+			if (httpXhr.status === 0) {
+				// https is failing as well...service is unavailable
+				self.Dispatcher.handleAppEvent({
+					name: "SERVICE_UNAVAILABLE",
+					status: httpXhr.status
+				});
+			}
+			// https did not work but http did without a network error
+			// => missing ssl exception for controller
+			self.Dispatcher.handleAppEvent({
+				name: "HTTPS_CERT_MISSING",
+				status: httpXhr.status
+			});
+		};
+
+		if (httpsXhr.status === 0) {
+			// https is unavailable, let's see if http works
+			return self.client.ping("controller", "http").then(handleSuccess).catch(handleError);
+		}
+		// an error code other than 0
+		self.Dispatcher.handleAppEvent({
+			name: "SERVICE_UNAVAILABLE",
+			status: httpsXhr.status
+		});
+	},
+
+	__catchSecurePingResponse: function(args, promise) {
+		var xhr = args[1];
+		if (xhr.status === 0) {
+			// We were not able to access the controller due to a network error (ssl, timeout)
+			// In order to understand what's happening, we have to switch to http.
+			this.Dispatcher.handleAppEvent({
+				name: "CONTROLLER_UNREACHABLE_FROM_HTTPS",
+				status: xhr.status
+			});
+			console.log(this);
+			promise.reject();
```

Comment:
[`return Promise.reject(new Error("..."))`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/reject) should work (the name of the above event is fine for the error string).